pull_infra.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
```

Comment:
Would it make sense to make this into a cmake script so we get Windows support?

Reply:
I would also like this functionality to be cross-platform. However, I think that CMake is not the right option here, since the sync should be unrelated to the configure-build-test steps. I will try to push a python version of it.

Reply:
With a little help from AI I created a python version and I run it successfully on windows with python 3.12 installed.

Reply:
You can write a full program in CMake and run it with `cmake -P arbitrary_file_name`. See the docs or `--help` for the `cmake` command for reference.

If you name the script `pull_infra`, make it executable, and add `#!/usr/bin/env cmake -P` as the shebang line, the user would just run it and not care what language it's written in. It would also be awkward to intentionally pull into a `CMakeLists.txt` on accident because the lack of a standard name for a CMake file.

Reply:
That being said, given we seem fine using precommit for SDLC purposes, I don't mind shipping other utilities in python. But it's true that a fairly simple workflow like this could be written as a standalone script in CMake with fewer dependencies for the users we're targeting. It would be harder to maintain, though, probably.

Reply:
I didn't know that I could run it with cmake like that. Anyways, I think that python is fine for now and it is not an overhead, since we already installed it in our environment due to precommit and also other C++ tooling like conan uses python.